### PR TITLE
Cmdlet to package dsc modules and configuration and publish on dsc pu…

### PR DESCRIPTION
**Pull request by NarineM**
Created a powershell script to package dsc modules and configuration documents and publish them on dsc enterprise pull server.

### CODE REVIEW

**Comment by KarolKaczmarek** [comment]
I think a better place to put it would be xDscUtils repo (https://github.com/powershell/xdscutils )

**Comment by KarolKaczmarek** [comment]
Also, please fix the test failures (replace tabs with spaces)
```
44There are tab in C:\projects\xpsdesiredstateconfiguration\DSCPullServer\PublishModulesAndMofsToPullServer.ps1. Use Fixer 'Get-TextFilesList $pwd | ConvertTo-SpaceIndentation'.
 45    [-] Uses spaces for indentation, not tabs 388ms 
 46      Expected: {0} 
 47      But was:  {1} 
 48      135:                 $totalTabsCount | Should Be 0 
 49      at <ScriptBlock>, C:\projects\xpsdesiredstateconfiguration\DscResource.Tests\Meta.Tests.ps1: line 135 
```

**Comment by NarineM** [comment]
@KarolKaczmarek , thanks for the review. converted tabs to spaces.

File: DSCPullServerSetup/PublishModulesAndMofsToPullServer.psd1 [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+GUID = 'e4685473-cdc7-4396-bfbf-eab80e6f029d'
+
+# Author of this module
+Author = 'Narine Mossikyan'
+
+# Company or vendor of this module
+CompanyName = 'Microsoft Corporation'
+
+# Copyright statement for this module
+Copyright = '(c) 2014 Microsoft Corporation. All rights reserved.'
```
Comment (TravisEz13) [inline DSCPullServerSetup/PublishModulesAndMofsToPullServer.psd1]: copyright date

File: DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1 [hunk 2]
```diff
@@ -1,11 +1,31 @@
+<#
+.Synopsis
+   Package DSC modules and mof configuration document and publish them on enterprise DSC pull server in the required format
+.DESCRIPTIO
```
Comment (TravisEz13) [inline DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1]: description is misspelled

File: DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1 [hunk 3]
```diff
@@ -108,17 +143,4 @@ function PublishMofDocuments
...
-# Generate the checkSum file for all the zip and mof files.
-New-DSCCheckSum $tempFolder -Force
-# Publish mof and modules to pull server repositories
-PublishModulesAndChecksum -source $tempFolder
-PublishMofDocuments -source $tempFolder
-#Deployment is complete!
-Remove-Item -Path $tempFolder -Recurse -Force -ErrorAction SilentlyContinue
-Write-Host "End deployment"
\ No newline at end of file
+}
```
Comment (TravisEz13) [inline DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1]: Files must end with a new line.  See [Style Guidelines, General Rule, #5](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md#general-rules)

File: DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1 [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    $mofRepository = "$env:ProgramFiles\WindowsPowerShell\DscService\Configuration"
+    if( (Get-Module ServerManager -ListAvailable) -and (Test-Path ($mofRepository)) )    
+    {
+        Copy-Item "$source\*.mof*" $mofRepository -Force -Verbose
+    }
+    else
+    {
+        Write-Host "Copying configuration(s) to pullserver configuration repository skipped because the machine is not a server sku or Pull server endpoint is not deployed." -Fore Yellow
+    } 
+}
```
Comment (TravisEz13) [inline DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1]: Files must end with a new line.  See [Style Guidelines, General Rule, #5](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md#general-rules)

File: DSCPullServerSetup/PublishModulesAndMofsToPullServer.psd1 [hunk 5]
```diff
@@ -0,0 +1,80 @@
+#
```
Comment (TravisEz13) [inline DSCPullServerSetup/PublishModulesAndMofsToPullServer.psd1]: Why do we need a separate module?

File: DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1 (hunk 2 above)
Comment (NarineM) [inline DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1]: Fixed.

File: DSCPullServerSetup/PublishModulesAndMofsToPullServer.psd1 (hunk 5 above)
Comment (NarineM) [inline DSCPullServerSetup/PublishModulesAndMofsToPullServer.psd1]: removed psd1 .

File: DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1 (hunk 4 above)
Comment (NarineM) [inline DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1]: Fixed.

**Pull request merged**